src/did/src/codec.rs

```diff
@@ -1,9 +1,17 @@
 use candid::{CandidType, Decode, Deserialize, Encode};
 
-pub fn encode(item: &impl CandidType) -> Vec<u8> {
+pub fn encode<T: CandidType>(item: &T) -> Vec<u8> {
```

Comment:
Just curious: what is the difference?

Reply:
They are nearly identical, I am not sure why I changed it